### PR TITLE
plugins/new-file-template: init module

### PR DESCRIPTION
**Pull request by jacekpoz**
no option to add templates because diniamo said so

they can be added by adding them to the runtime path under `lua/templates/`

### CODE REVIEW

File: modules/plugins/utility/new-file-template/default.nix [hunk 1]
```diff
@@ -0,0 +1,6 @@
+_: {
```
Comment (diniamo) [inline modules/plugins/utility/new-file-template/default.nix:1]: Remove `_: `

**Review by diniamo — CHANGES_REQUESTED** [review]

**Comment by diniamo** [comment]
It would also be nice to not include irrelevant stuff in this pr, and to document adding templates somewhere, because it's not exactly clear right now.

**Comment by jacekpoz** [comment]
adding templates is pretty clear from the docs of the plugin

File: modules/plugins/utility/new-file-template/new-file-template.nix [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+}: let
+  inherit (lib.options) mkOption;
+  inherit (lib.types) attrsOf bool listOf str;
+  inherit (lib.nvim.types) mkPluginSetupOption;
+in {
+  options.vim.utility.new-file-template = {
+    enable = mkOption {
+      type = bool;
+      default = false;
+      description = "new-file-template.nvim: Automatically insert a template on new files in neovim";
```
Comment (NotAShelf) [inline modules/plugins/utility/new-file-template/new-file-template.nix:13]: I'd prefer an option to add templates but if we are not doing that, lets either
a. tell people in the enable option how to add templates
b. add a documentation page for the plugin and link it here

**Review by NotAShelf — CHANGES_REQUESTED** [review]

File: modules/plugins/utility/new-file-template/new-file-template.nix (hunk 2 above)
Comment (jacekpoz) [reply modules/plugins/utility/new-file-template/new-file-template.nix]: check the newest version are these docs ok

File: modules/plugins/utility/new-file-template/new-file-template.nix [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+      disableFiletype = mkOption {
+        type = listOf str;
+        default = [];
+        description = "Disable templates for specific filetypes (only disables default templates, user templates will still work)";
+      };
+
+      disableSpecific = mkOption {
+        type = attrsOf (listOf str);
+        default = {};
+        description = "Disable specific regexp for the default templates. Example: { ruby = [ \".*\" ]; }";
```
Comment (NotAShelf) [inline modules/plugins/utility/new-file-template/new-file-template.nix:42]:
```suggestion
        description = "Disable specific regexp for the default templates.";
```

File: modules/plugins/utility/new-file-template/new-file-template.nix [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+      disableAutocmd = mkOption {
+        type = bool;
+        default = false;
+        description = "Disable the autocmd that creates the template";
+      };
+
+      disableFiletype = mkOption {
+        type = listOf str;
+        default = [];
+        description = "Disable templates for specific filetypes (only disables default templates, user templates will still work)";
```
Comment (NotAShelf) [inline modules/plugins/utility/new-file-template/new-file-template.nix:36]: Try to limit characters per line to 80 here.

File: modules/plugins/utility/new-file-template/new-file-template.nix [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+  inherit (lib.types) attrsOf bool listOf str;
+  inherit (lib.nvim.types) mkPluginSetupOption;
+in {
+  options.vim.utility.new-file-template = {
+    enable = mkOption {
+      type = bool;
+      default = false;
+      description = ''
+        new-file-template.nvim: Automatically insert a template on new files in neovim.
+        To add custom templates add a directory containing `lua/templates/*.lua` to `vim.additionalRuntimePaths`.
```
Comment (NotAShelf) [inline modules/plugins/utility/new-file-template/new-file-template.nix:15]: Wrap this line in 
```
::: {.note}

:::
```

to mark this line as a note annotation in the documentation.

Also, limit characters per line to 80 and use marked links for the URL.

File: modules/plugins/utility/new-file-template/new-file-template.nix [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+        type = attrsOf (listOf str);
+        default = {};
+        description = "Disable specific regexp for the default templates. Example: { ruby = [ \".*\" ]; }";
+        example = "{ ruby = [\".*\"]; }";
+      };
+
+      suffixAsFiletype = mkOption {
+        type = bool;
+        default = false;
+        description = "Use suffix of filename rather than vim.bo.filetype as filetype";
```
Comment (NotAShelf) [inline modules/plugins/utility/new-file-template/new-file-template.nix:49]: Put `vim.bo.filetype` into inline code mode:

```
`vim.bo.filetype`
```

**Review by NotAShelf — CHANGES_REQUESTED** [review]
Some "minor" nits.

**Review by NotAShelf — DISMISSED** [review]

**Pull request merged**